lib/view/login_screen.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+import 'package:the_quran/controller/authentication_controller.dart';
+import 'package:the_quran/model/authentication_model.dart';
+import 'package:the_quran/view/home_page.dart';
+import 'package:the_quran/view/forgot_password_screen.dart';
+import 'package:the_quran/view/sign_up_screen.dart';
+
+class LoginScreen extends StatelessWidget {
+  final AuthenticationController authController;
+
+  LoginScreen({required this.authController});
+
+  @override
+  Widget build(BuildContext context) {
+    TextEditingController emailController = TextEditingController();
+    TextEditingController passwordController = TextEditingController();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Log In'),
+      ),
+      body: Padding(
+        padding: EdgeInsets.all(16),
+        child: Column(
+          children: [
+            TextFormField(
+              controller: emailController,
+              decoration: InputDecoration(labelText: 'Email'),
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter your email';
+                }
+                if (!RegExp(r'^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$')
+                    .hasMatch(value)) {
+                  return 'Please enter a valid email';
+                }
+                return null;
+              },
+            ),
+            TextFormField(
+              controller: passwordController,
+              decoration: InputDecoration(labelText: 'Password'),
+              obscureText: true,
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter your password';
+                }
+                if (value.length < 8) {
+                  return 'Password must be at least 8 characters long';
+                }
+                if (!RegExp(
+                        r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$')
+                    .hasMatch(value)) {
+                  return 'Password must contain at least one uppercase letter, one lowercase letter, one digit, and one special character';
```

Comment:
No need for this in the login.

Reply:
I think It will improve performance to make sure that the user enters a valid email before calling a function otherwise the function will get called and require resources just to return invalid email

Reply:
okay for now, just validate the email field with regex, and for the password only check the length (signup screen only).